pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -11684,3 +11686,121 @@ func TestCloudstorageParallelCompression(t *testing.T) {
 		}
 	})
 }
+
+func TestChangefeedBareFullProtobuf(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+		sqlDB := sqlutils.MakeSQLRunner(s.DB)
+
+		sqlDB.Exec(t, `
+			CREATE TABLE pricing (
+				id INT PRIMARY KEY,
+				name STRING,
+				discount FLOAT,
+				tax DECIMAL,
+				options STRING[],
+				details JSONB 
+			)`)
+
+		sqlDB.Exec(t, `
+			INSERT INTO pricing VALUES
+				(1, 'Chair', 15.75, 2.500, ARRAY['Brown', 'Black'], '{"material": "oak", "height": 54}'),
+				(2, 'Table', 20.00, 1.23456789, ARRAY['Brown', 'Black'], '{"material": "birch", "height": 62}')
+		`)
+
+		pricingFeed := feed(t, f, `CREATE CHANGEFEED FOR pricing WITH envelope='bare', format='protobuf', mvcc_timestamp, updated, key_in_value, topic_in_value`)
+		defer closeFeed(t, pricingFeed)
+
+		type rowExpectations struct {
+			name     string
+			discount float64
+			tax      *apd.Decimal
+			options  []string
+			details  json.JSON
+		}
+
+		decimal1 := new(apd.Decimal)
+		decimal2 := new(apd.Decimal)
+		var err error
+		_, _, err = decimal1.SetString("2.500")
+		require.NoError(t, err)
+		_, _, err = decimal2.SetString("1.23456789")
+		require.NoError(t, err)
+
+		json1, err := json.ParseJSON(`{"material": "oak", "height": 54}`)
+		require.NoError(t, err)
+		json2, err := json.ParseJSON(`{"material": "birch", "height": 62}`)
+		require.NoError(t, err)
+
+		expected := map[int64]rowExpectations{
+			1: {
+				name:     "Chair",
+				discount: 15.75,
+				tax:      decimal1,
+				options:  []string{"Brown", "Black"},
+				details:  json1,
+			},
+			2: {
+				name:     "Table",
+				discount: 20.00,
+				tax:      decimal2,
+				options:  []string{"Brown", "Black"},
+				details:  json2,
+			},
+		}
+
+		received := make(map[int64]bool)
+		for len(received) < len(expected) {
+			row, err := pricingFeed.Next()
```

Comment:
I'd prefer you to use the `assertPayloads` helper for this stuff. That will require you to write a pb -> json decoder inside it (similar to what is there for avro etc). please either do that here or make a todo with issue for it